### PR TITLE
fix(discord): ignore invalid poll payloads

### PR DESCRIPTION
**Pull request by deanfluencebot**
## Summary
- ignore invalid Discord poll payloads in rich replies
- keep plain text replies working even when a malformed optional poll object is present
- add a unit test covering empty and incomplete poll payloads

## Why
Some reply payloads can include an empty poll stub even when the intent is just a normal text reply. Discord rejects those malformed polls, which causes the whole rich message send to fail.

This change validates poll payloads before attaching them to Discord messages. Invalid polls are skipped instead of breaking delivery.

## Testing
- added unit test coverage for invalid and valid poll payloads

> [!NOTE]
> This PR adds poll validation to the Discord adapter. A new `is_valid_poll()` function checks that polls have a non-empty question and at least 2 answers, filtering them at the point of attachment in two locations. Invalid polls are silently skipped with a debug log. The unit test covers empty, single-answer, and valid poll payloads.
>
> <sub>Written by [Tembo](https://app.tembo.io) for commit [badfb61](https://github.com/spacedriveapp/spacebot/commit/badfb611f6f5b86b5424cbab0184f94effce3144). This will update automatically on new commits.</sub>

### CODE REVIEW

File: src/messaging/discord.rs [hunk 1]
```diff
@@ -1176,6 +1176,26 @@ fn build_poll(
...
+        tracing::debug!(
+            question_len = question.len(),
+            answer_count,
+            "skipping invalid discord poll payload"
+        );
+        return false;
+    }
+
+    true
+}
```
[the remark this answers is not in the document]
Comment (deanfluencebot) [reply src/messaging/discord.rs:1201]: Addressed in c42dae2. I added a shared `normalized_poll_answers` helper and now both `is_valid_poll` and `build_poll` use the same trim/filter/take-10 path. I also added regression coverage for sparse answers and the first-10-normalized behavior. Targeted test run: `cargo test -q test_invalid_poll_is_rejected --manifest-path /Users/botfluence/spacebot/Cargo.toml`.

File: src/messaging/discord.rs [hunk 2]
```diff
@@ -1176,6 +1176,26 @@ fn build_poll(
     p
 }
 
+fn is_valid_poll(poll: &crate::Poll) -> bool {
+    let question = poll.question.trim();
+    let answer_count = poll
```
[the remark this answers is not in the document]
Comment (deanfluencebot) [reply src/messaging/discord.rs]: Addressed in c42dae2. The validation now uses the same normalized answer set as serialization via `normalized_poll_answers`, so we no longer count answers that would be dropped or skipped before sending to Discord.
